src/videoframeprocessor/DefaultVideoFrameProcessorPipeline.ts

```diff
@@ -45,8 +45,52 @@ export default class DefaultVideoFrameProcessorPipeline implements VideoFramePro
 
   private hasStarted: boolean = false;
   private lastTimeOut: ReturnType<typeof setTimeout> | undefined;
+  // Timer on worker thread to overcome background throttling
+  private workerTimer: Worker;
+  private isWorkerTimerSupported: boolean = true;
 
-  constructor(private logger: Logger, private stages: VideoFrameProcessor[]) {}
+  constructor(private logger: Logger, private stages: VideoFrameProcessor[]) {
+    this.workerTimer = this.createWorkerTimer();
+  }
+
+  /**
+   * Create a timer that exists within a web worker. This timer will be used to
+   * retrigger the process call whenever time expires.
+   */
+  private createWorkerTimer(): Worker {
```

Comment:
nit: This function could just fail and return undefined, which the calling code can use as an alternative to `isWorkerTimerSupported`. Then it could be moved outside of the class definition, in case we need to use this elsewhere.

Reply:
If you really want to have some fun and make this testable, you could make some `VideoFrameProcessorPipelineTimer` interface to make this easier to test :)

Reply:
Agree with Henry, this could be turned into a `util` and unit tested in itself.

Reply:
Is it legal to resolve your own comments by pushing commits to someone else's PR? 🤔